src/main/java/com/sintern/api/OpenInternPositionController.java

```diff
@@ -1,37 +1,66 @@
 package com.sintern.api;
 
+import com.sintern.domain.dto.OpenInternPositionDTO;
+import com.sintern.domain.entity.Company;
+import com.sintern.domain.entity.CompanyLogo;
 import com.sintern.domain.entity.Domain;
+import com.sintern.domain.entity.OpenInternPosition;
 import com.sintern.domain.enums.DomainType;
-import com.sintern.domain.dto.OpenInternPositionDTO;
-import com.sintern.service.OpenInternPositionService;
+import com.sintern.service.LogoService;
+import com.sintern.service.OpenInternPositionServiceImpl;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
 
 @CrossOrigin
 @RestController
 @RequestMapping("/open-positions")
 public class OpenInternPositionController {
-    private final OpenInternPositionService openInternPositionService;
+    private final OpenInternPositionServiceImpl openInternPositionService;
```

Comment:
Why did you change this? We should use **interfaces**, not **concrete implementations**.

Reply:
My inattention. There is no interface. I added it and refactored it automatically.